// Copyright 2015 gRPC authors.
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//     http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.

syntax = "proto3";

option java_multiple_files = true;
option java_package = "io.grpc.examples.dispersion";
option java_outer_classname = "DispersionProto";
option objc_class_prefix = "DISP";

package dispersion;

// The dispersion service definition.
service Dispersion {
  // Send results of dispersion calculations.
  rpc GetDispersionCalculations (DispersionRequest) returns (DispersionReply) {}
}

// The request message containing dispersion calculation results.
message DispersionRequest {
  uint32 shard = 1;
  float mean = 2;
  float dispersion = 3;
  float sum = 4;
  uint32 count = 5;
}

// The response message containing the greetings
message DispersionReply {
  uint32 result = 1;
}
